### PR TITLE
fix(Networking): fix unable to edit Host Aliases configuration of a workload pod

### LINKED ISSUE #12991: Unable to edit Host Aliases configuration of a workload pod
**Setup**
- Rancher version: v2.10.1/v2.9.x
- Browser type & version: 131.0.6778.205

**Describe the bug**

1. When you click the Add button to add Host Aliases, they are automatically deleted immediately.
2. When editing a workload, if the workload already has Host Aliases configured, clicking the Add button to add Host Aliases will report a js error.

**To Reproduce**

1. Go to the create workload page
2. Check the `Pod -> Networking` tab section
3. In the Host Aliases section, click the `Add Aliases` button

**Result**

**Expected Result**

When host Aliases are added, the corresponding form is displayed.

**Screenshots**

workload_hostalias_error
workload_hostalias_error1

**Additional context**

### PR DESCRIPTION
**Pull request by ly5156**
### Summary
Fixes #12991

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by cnotv** [comment]
@richard-cox @ly5156 pipeline is blocked.

**Comment by richard-cox** [comment]
It had a flacky test, which given a couple of re-runs now pass

**Comment by ly5156** [comment]
@cnotv 
I have added a unit test file(Networking.test.ts)

**Comment by cnotv** [comment]
@richard-cox please communicate issues of the new pipeline `Validate Pull Request Description` to the reported next time, thanks.

**Comment by cnotv** [comment]
> @cnotv I have added a unit test file(`Networking.test.ts`)

I am glad to see it helped to find an edge case 🍻

**Review by cnotv — APPROVED** [review]
Added just one point, but not a blocker.
I have added suggestions for the unit tests, but are not an issue at all as for tradition we do not block for this.

LGTM

File: shell/components/form/__tests__/Networking.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    addButton.trigger('click');
+    await nextTick();
+
+    keyInput = hostAliases.find('[data-testid="input-kv-item-key-0"]');
+    valueInput = hostAliases.find('[data-testid="kv-item-value-0"]');
+
+    expect(keyInput.exists()).toBe(true);
+    expect(valueInput.exists()).toBe(true);
+
+    await new Promise((resolve) => {
```
[the remark this answers is not in the document]
Comment (cnotv) [reply shell/components/form/__tests__/Networking.test.ts]: Oh right, that was the debounce, nice catch 👍

**Review by cnotv — APPROVED** [review]

**Comment by richard-cox** [comment]
@ly5156 This unfortunately needs a rebase. @cnotv if after that the PR looks good can you approve and merge?

**Review by cnotv — CHANGES_REQUESTED** [review]
Unit tests seems failing: https://github.com/rancher/dashboard/actions/runs/13679791572/job/38249039235?pr=12992

Taking the chance we are addressing unit tests, a format AAA would be preferred ([doc example](https://extensions.rancher.io/internal/testing/unit-test#aaa-pattern)) although we do not enforce it in our repository.

Screenshot 2025-03-05 at 17 00 30

**Comment by ly5156** [comment]
I've modified the unit tests code according to AAA format，But I'm not sure if I'll still report the error, because running this unit tests code locally passes it

**Review by cnotv — DISMISSED** [review]
LGTM

**Comment by eva-vashkevich** [comment]
Hi @ly5156 , could you please rebase tis PR one more time? Thank you

**Comment by richard-cox** [comment]
We've missed the deadline on this one. @cnotv can you take a look early in 2.13.0?

**Comment by cnotv** [comment]
> We've missed the deadline on this one. @cnotv can you take a look early in 2.13.0?

Yeah SCC took priority, we should have reassigned it.

**Comment by ly5156** [comment]
> Hi @ly5156 , could you please rebase tis PR one more time? Thank you

@eva-vashkevich

I have rebase this PR
